src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/SocketOutput.cs

```diff
@@ -673,6 +673,7 @@ private void DoDisconnectIfNeeded()
                 Self._socket.Dispose();
                 Self._connection.OnSocketClosed();
                 Self._log.ConnectionStop(Self._connectionId);
+                KestrelEventSource.Log.ConnectionStop(Self._connection);
```

Comment:
Normally I would put stuff like this in the Connection class. So Connection.OnSocketClosed() would probably be better.